src/components/Header.tsx

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios';
+import { Component } from 'react';
+import { ISearchState } from '../App';
+
+interface Props {
+  searchState: ISearchState;
+  updateSearchState: (newState: ISearchState) => void;
+}
+
+class Header extends Component<Props> {
+  handleSearch() {
+    const { searchValue } = this.props.searchState;
+
+    const apiEndpoint =
+      searchValue === ''
+        ? `https://swapi.dev/api/planets/`
+        : `https://swapi.dev/api/planets/?search=${searchValue}`;
+
+    axios
+      .get(apiEndpoint)
+      .then((response) => {
+        this.props.updateSearchState({
+          searchValue: searchValue,
+          searchResults: response.data.results,
+        });
+      })
+      .catch(() => {
+        console.log('Not Found');
+      });
+    localStorage.setItem('prevSearchValue', searchValue);
```

Comment:
Task require to have search field in one component and render in other one. So I decided to bring state of entered search value to common parent component. I also tried to use React Context instead of passing props, but did not get luck with it.